src/execution_tree/primitives/row_set.cpp

```diff
@@ -0,0 +1,341 @@
+
+        protected:
+            using arg_type = ir::node_data<double>;
+            using args_type = std::vector<arg_type>;
+
+            using storage0d_type = typename arg_type::storage0d_type;
+            using storage1d_type = typename arg_type::storage1d_type;
+            using storage2d_type = typename arg_type::storage2d_type;
+
+            std::vector<int> create_list_row_set(
+                int start, int stop, int step, int array_length) const
+            {
+                HPX_ASSERT(step != 0);
+                auto actual_start = 0;
+                auto actual_stop = 0;
+
+                if (start >= 0)
+                {
+                    actual_start = start;
+                }
+                else //(start < 0)
+                {
+                    actual_start = array_length + start;
+                }
+
+                if (stop >= 0)
+                {
+                    actual_stop = stop;
+                }
+                else //(stop < 0)
+                {
+                    actual_stop = array_length + stop;
+                }
+
+                std::vector<int> result;
+
+                if (step > 0)
+                {
+                    result.reserve((actual_stop - actual_start + step)/step);
+                    for (int i = actual_start; i < actual_stop; i += step)
+                    {
+                        result.push_back(i);
+                    }
+                }
+                else //(step < 0)
+                {
+                    result.reserve((actual_start - actual_stop - step)/(-step));
+                    for (int i = actual_start; i > actual_stop; i += step)
+                    {
+                        result.push_back(i);
+                    }
+                }
+
+                if (result.empty())
+                {
+                    HPX_THROW_EXCEPTION(hpx::bad_parameter,
+                        "phylanx::execution_tree::primitives::"
+                        "row_set_operation::create_list_row_set",
+                        generate_error_message(
+                            "Set will produce empty result, please check your "
+                            "parameters",
+                            name_, codename_));
+                }
+
+                return result;
+            }
+
+            primitive_argument_type row_set0d(args_type&& args) const
+            {
+                HPX_THROW_EXCEPTION(hpx::bad_parameter,
+                    "phylanx::execution_tree::primitives::"
+                    "row_set_operation::row_set_0d",
+                    generate_error_message(
+                        "use store operation for setting value to a variable",
+                        name_, codename_));
+            }
+
+            primitive_argument_type row_set1d(args_type&& args) const
+            {
+                auto row_start = args[1].scalar();
+                auto row_stop = args[2].scalar();
+                int step = args[3].scalar();
```

Comment:
Usually `int` can't fully represent the value range of `double`, so this might go wrong.